scripts/run_docker.ps1

```diff
@@ -1,18 +1,44 @@
-$is_exited
-$is_not_first
-while(!$is_exited) {
-    $docker_command = -join("docker run -it -v $HOME/ChangeUp:/root/ChangeUp raiderrobotics/container-registry:rr-noetic-base sh -c ", "'cd /root/ChangeUp;", $args[0], "'");
-    Invoke-Expression -Command $docker_command
-    if (!(docker ps -a -q -f ancestor=raiderrobotics/container-registry:rr-noetic-base -f status=exited)) {
-        if(!($is_not_first)) {
-            C:/Program` Files/Docker/Docker/Docker` Desktop.exe -WindowsStyle Minimized
-            $is_not_first = "True"
+$docker_command = "docker run -it"
+$docker_vol_location = -join(" -v ", $PSScriptRoot.Replace("/scripts", ""), ":/root/ChangeUp raiderrobotics/container-registry:rr-noetic-base")
+$docker_container_command = -join(" sh -c 'cd /root/ChangeUp;", $args[0], "'")
+
+# $IsWindows doesn't work because powershell 5.1 is too old
+if ([System.Environment]::OSVersion.Platform -eq "Win32NT") {
+    $is_not_first
+    $is_exited
+    $joined_cmd = -join($docker_command, $docker_vol_location, $docker_container_command)
+    while(!$is_exited) {
+	# run docker contianer
+        Invoke-Expression -Command $joined_cmd
+	# check if container has exited to see if it ran successfully (will fail if docker desktop is not running)
+        if (!(docker ps -a -q -f ancestor=raiderrobotics/container-registry:rr-noetic-base -f status=exited)) {
+	    # attempt to launch docker desktop only once
+            if(!($is_not_first)) {
+                C:/Program` Files/Docker/Docker/Docker` Desktop.exe -WindowsStyle Minimized
+                $is_not_first = "True"
+            }
+            Write-Host "Docker Desktop Not Running"
+            Start-Sleep -Seconds 1
+        }
+        else {
+	    # remove container after it exits
+            docker rm $(docker ps --filter status=exited -q)

```

Comment:
Would this remove stopped containers if they are not ours? Ex. What if the user has a stopped container that we don't want to interfere with; would we end up removing it?

Reply:
Yes, currently it removes all containers with the exited status, including ones that are not ours. I recall recognizing the issue, but can't recall why I didn't fix it. I should just be able to replace the line's `docker ps` command with the one on line 14. I'll do that at the controls meeting tonight.